tests/50federation/30room-join.pl

```diff
@@ -953,7 +953,117 @@ sub assert_is_valid_pdu {
 
             # XXX currently synapse fails with a 500 here. I'm not really convinced that's
             # a thing we want to enforce, but we don't really have a specced way to say
-            # "a remote server did something wierd".
+            # "a remote server did something weird".
+            Future->done(1);
+         }),
+      )
+   };
+
+test "Outbound federation event from other origin",
```

Comment:
some more words about what we're testing here, maybe?

Reply:
I updated this. 👍